Tests/Unit/Controller/FrontEndEditorControllerTest.php

```diff
@@ -0,0 +1,117 @@
+<?php
+
+declare(strict_types=1);
+
+namespace TTN\Tea\Tests\Unit\Controller;
+
+use PHPUnit\Framework\MockObject\MockObject;
+use TTN\Tea\Controller\FrontEndEditorController;
+use TTN\Tea\Domain\Repository\Product\TeaRepository;
+use TYPO3\CMS\Core\Context\Context;
+use TYPO3\CMS\Core\Context\UserAspect;
+use TYPO3\CMS\Core\Http\HtmlResponse;
+use TYPO3\CMS\Core\Information\Typo3Version;
+use TYPO3\CMS\Core\Utility\GeneralUtility;
+use TYPO3\CMS\Extbase\Mvc\Controller\ActionController;
+use TYPO3\CMS\Extbase\Persistence\QueryResultInterface;
+use TYPO3\CMS\Fluid\View\TemplateView;
+use TYPO3\TestingFramework\Core\AccessibleObjectInterface;
+use TYPO3\TestingFramework\Core\Unit\UnitTestCase;
+
+/**
+ * @covers \TTN\Tea\Controller\FrontEndEditorController
+ */
+final class FrontEndEditorControllerTest extends UnitTestCase
+{
+    protected bool $resetSingletonInstances = true;
+
+    /**
+     * @var FrontEndEditorController&MockObject&AccessibleObjectInterface
+     */
+    private FrontEndEditorController $subject;
+
+    /**
+     * @var TemplateView&MockObject
+     */
+    private TemplateView $viewMock;
+
+    /**
+     * @var TeaRepository&MockObject
+     */
+    private TeaRepository $teaRepositoryMock;
+
+    private Context $context;
+
+    protected function setUp(): void
+    {
+        parent::setUp();
+
+        // We need to create an accessible mock in order to be able to set the protected `view`.
+        $methodsToMock = ['htmlResponse', 'redirect', 'redirectToUri'];
+        if ((new Typo3Version())->getMajorVersion() <= 11) {
+            $methodsToMock[] = 'forward';
+        }
+        $this->subject = $this->getAccessibleMock(FrontEndEditorController::class, $methodsToMock);
+
+        $this->viewMock = $this->createMock(TemplateView::class);
+        $this->subject->_set('view', $this->viewMock);
+
+        $this->teaRepositoryMock = $this->getMockBuilder(TeaRepository::class)->disableOriginalConstructor()->getMock();
```

Comment:
You are the export regarding unit testing :) So I use this opportunity to ask you what's the benefit of this approach over `->createStub()`?

Reply:
`createStub()` indeed is the better choice for our mocks here. (I had completely missed out on this method when it got introduced.) Thanks for pointing this out!